replacy/__init__.py

```diff
@@ -136,28 +143,48 @@ def get_predicates(self, match_hooks) -> List[Callable]:
             predicates.append(pred)
         return predicates
 
-    def inflect_suggestion(self, pre_suggestion, doc, start, end, match_name):
+    def process_suggestions(self, pre_suggestion, doc, start, end, match_name):
         """
+        Perform inflection and replace references to the matched token
         example of pattern and pre_suggestion
         pattern: "LEMMA": "chock", "TEMPLATE_ID": 1
         pre_suggestion: "TEXT": "chalk", "FROM_TEMPLATE_ID": 1
         inflect suggestion "chalk" according to form of "cholk" from patterns
         """
         text_list = []
         for item in pre_suggestion:
-            text = item["TEXT"]
+            try:
+                text = item["TEXT"]
+            except KeyError:
+                ref = int(item["PATTERN_REF"])
+                refd_token = doc[start + ref]
             changed_text = None
 
             # check if inflect
             if "FROM_TEMPLATE_ID" in item:
                 template_id = item["FROM_TEMPLATE_ID"]
                 index = None
-                for i, token in enumerate(self.match_dict[match_name]["patterns"]):
-                    if "TEMPLATE_ID" in token and token["TEMPLATE_ID"] == template_id:
+                for i, token in enumerate(
+                    self.match_dict[match_name]["patterns"]
+                ):
+                    if (
+                        "TEMPLATE_ID" in token
+                        and token["TEMPLATE_ID"] == template_id
+                    ):
                         index = i
                         break
                 if index is not None:
-                    changed_text = self.inflector.inflect(doc, text, start + index)
+                    changed_text = self.inflector.auto_inflect(
+                        doc, text, start + index
```

Comment:
This assumes `PATTERN REF` is always disjoint from `FROM TEMPLATE_ID`.
What if I want to copy a token from a pattern with inflection copied from another token? Or insert TEXT with `INFLECTION`?
ie would be nice to extend functionality of previous patterns instead of adding new types of patterns which lack functionality of the first.
Otherwise `lt-patterns` can only change inflections.

Reply:
So, I think if they want to do `TEXT + INFLECTION` they can just do that lookup before writing the pattern (`run + VBN` is `ran`, just look that up), but you're right about `PATTERN_REF` and `FROM_TEMPLATE_ID`. I did consider this.

The reason I thought this was okay is that `PATTERN_REF` can hit any token you want, as in, there is no functionality that you can't build with this implementation (though it's not as clean as I'd like) that you could build if the combo of `PATTERN_REF` and `FROM_TEMPLATE_ID` were supported

Reply:
> Would be nice to extend replaCy functionality instead of adding new.

does this comment mean you think we shouldn't be able to change the replaCy syntax?

Reply:
It means I prefer adding new features to tokens that can be used jointly instead of adding a separate case. There is no `should/shouldn't`. Just aesthetics.

Comparison:
- copy token or inflection from any token from the pattern
- vs. error. no, you can just set inflection if you are copying a token, you can't copy inflection

I imagine an interface when you can freely point arrows to construct suggestions from patterns. Case 2 would be a strange restriction.

Current state is already cool though, so I am fine with considering extensions later.